nav2_behavior_tree/plugins/control/round_robin_node.cpp

```diff
@@ -37,52 +37,57 @@ BT::NodeStatus RoundRobinNode::tick()
 
   setStatus(BT::NodeStatus::RUNNING);
 
-  while (num_failed_children_ < num_children) {
+  while (num_failed_children_ + num_skipped_children_ < num_children) {
     TreeNode * child_node = children_nodes_[current_child_idx_];
     const BT::NodeStatus child_status = child_node->executeTick();
 
+    if (child_status != BT::NodeStatus::RUNNING) {
+      // Increment index and wrap around to the first child
+      if (++current_child_idx_ == num_children) {
+        current_child_idx_ = 0;
+      }
+    }
+
     switch (child_status) {
       case BT::NodeStatus::SUCCESS:
         {
-          // Wrap around to the first child
-          if (++current_child_idx_ >= num_children) {
-            current_child_idx_ = 0;
-          }
           num_failed_children_ = 0;
+          num_skipped_children_ = 0;
```

Comment:
Looking at this, it seems like `num_skipped_children_` can be a method variable not a class member that needs to be managed between ticks. We iterate through each children and only return `SKIPPED` if all are skipped, so all of the state needed to manage whether to return skipped is managed in this node's tick in a single snapshot

Reply:
Correct. Do you want me to change it?

Reply:
Please - one less thing for me or someone else to mismanage later